### PR TITLE
feat(public-accessibility-logic): Add AKS and AWS Lambda

### PR DESCRIPTION
**Pull request by Flowtter**
Add documentation for AKS and AWS lambda.
Following https://github.com/DataDog/dd-source/pull/83897 https://github.com/DataDog/dd-source/pull/83596

### CODE REVIEW

**Review by FredMoster — APPROVED** [review]

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 1]
```diff
@@ -202,6 +202,16 @@ An [SQS Queue][32] (`aws_sqs_queue`) is considered publicly accessible if:
 
 See [Amazon SQS security best practices][33] for more information about public SQS queues.
 
+### Amazon Lambda Function
+
+A [lambda function][58] (`aws_lambda_function`) is considered publicly accessible if:
```
Comment (buraizu) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:207]:
```suggestion
A [Lambda function][58] (`aws_lambda_function`) is considered publicly accessible if:
```

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 2]
```diff
@@ -202,6 +202,16 @@ An [SQS Queue][32] (`aws_sqs_queue`) is considered publicly accessible if:
 
 See [Amazon SQS security best practices][33] for more information about public SQS queues.
 
+### Amazon Lambda Function
```
Comment (buraizu) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:205]:
```suggestion
### AWS Lambda function
```

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 3]
```diff
@@ -202,6 +202,16 @@ An [SQS Queue][32] (`aws_sqs_queue`) is considered publicly accessible if:
...
+
+A [lambda function][58] (`aws_lambda_function`) is considered publicly accessible if:
+
+| **Criteria** | **Explanation** |
+|--------------|-----------------|
+|The function has a policy that allows any principal (principal_policy or principal_aws set to `"*"`) to perform actions unconditionally.|This setting makes the function accessible to everyone in the world or to any authenticated AWS user.|
```
Comment (buraizu) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:211]:
```suggestion
|The function has a policy that allows any principal (`principal_policy` or `principal_aws`) set to `"*"`. |This setting makes the function accessible to everyone in the world or to any authenticated AWS user.|
```

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 4]
```diff
@@ -202,6 +202,16 @@ An [SQS Queue][32] (`aws_sqs_queue`) is considered publicly accessible if:
...
+
+See [AWS Lambda documentation][59] for more information about public Lambda functions.
```
Comment (buraizu) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:213]:
```suggestion
See [Best practices for working with AWS Lambda functions][59] for more information about public Lambda functions.
```

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 5]
```diff
@@ -251,6 +261,17 @@ A Storage Blob Container (`azure_storage_blob_container`) is considered publicly
 
 To learn more about disallowing blob public access on Azure Storage accounts, see [Choose to allow or disallow blob public access on Azure Storage accounts][46].
 
+### Azure AKS Cluster
```
Comment (buraizu) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:264]:
```suggestion
### Azure Kubernetes Service (AKS) cluster
```

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 6]
```diff
@@ -251,6 +261,17 @@ A Storage Blob Container (`azure_storage_blob_container`) is considered publicly
...
+
+An [AKS cluster][60] (`azure_aks_cluster`) is considered publicly accessible if:
+
+| **Criteria** | **Explanation** |
+|--------------|-----------------|
+|`enable_private_cluster` is set to `false` in the cluster's configuration.|This setting makes the cluster publicly accessible when combined with an open public CIDR. |
+|The cluster's `authorized_ip_ranges` contains an open CIDR block (`"0.0.0.0/0"`) or is unset.|You can limit the CIDR blocks that can access the public endpoint of the AKS cluster. An open CIDR block means anyone on the internet can access the endpoint.|
```
Comment (buraizu) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:271]:
```suggestion
|The cluster's `authorized_ip_ranges` contains an open CIDR block (`"0.0.0.0/0"`) or is unset.|An open CIDR block means anyone on the internet can access the endpoint.|
```

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 7]
```diff
@@ -251,6 +261,17 @@ A Storage Blob Container (`azure_storage_blob_container`) is considered publicly
...
+
+See [Azure Kubernetes Service (AKS) clusters][61] for more information on public AKS clusters.
```
Comment (buraizu) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:273]:
```suggestion
See [AKS best practices][61] for more information on public AKS clusters.
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks, just requesting some minor edits, but let me know if you have any feedback

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md (hunk 1 above)
Comment (Flowtter) [reply content/en/security/cloud_security_management/guide/public-accessibility-logic.md:207]: Following the nomenclature, I should capitalize both letters.

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md (hunk 4 above)
Comment (Flowtter) [reply content/en/security/cloud_security_management/guide/public-accessibility-logic.md]: Great changes.
I capitalized function though

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md (hunk 2 above)
Comment (buraizu) [reply content/en/security/cloud_security_management/guide/public-accessibility-logic.md:205]: Thanks for following up. Could you clarify what nomenclature you're referring to?

For additional context about why I've made this request:
 - The [documentation style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#headers) requests that level-3 headers use sentence case
 - `Lambda function` is used in AWS documentation (see [examples on this page](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html))
 - `Lambda function` is what's used in other pages of the documentation site (see [examples on this page](https://docs.datadoghq.com/serverless/aws_lambda/installation/python/?tab=datadogcli))

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 8]
```diff
@@ -202,6 +202,16 @@ An [SQS Queue][32] (`aws_sqs_queue`) is considered publicly accessible if:
 
 See [Amazon SQS security best practices][33] for more information about public SQS queues.
 
+### Amazon Lambda Function
+
+A [Lambda Function][58] (`aws_lambda_function`) is considered publicly accessible if:
```
Comment (buraizu) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:207]:
```suggestion
A [Lambda function][58] (`aws_lambda_function`) is considered publicly accessible if:
```

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md [hunk 9]
```diff
@@ -202,6 +202,16 @@ An [SQS Queue][32] (`aws_sqs_queue`) is considered publicly accessible if:
...
+
+| **Criteria** | **Explanation** |
+|--------------|-----------------|
+|The function has a policy that allows any principal (`principal_policy` or `principal_aws`) set to `"*"`. |This setting makes the function accessible to everyone in the world or to any authenticated AWS user.|
+
+See [Best practices for working with AWS Lambda Function][59] for more information about public Lambda functions.
```
Comment (buraizu) [inline content/en/security/cloud_security_management/guide/public-accessibility-logic.md:213]:
```suggestion
See [Best practices for working with AWS Lambda function][59] for more information about public Lambda functions.
```

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md (hunk 1 above)
Comment (buraizu) [reply content/en/security/cloud_security_management/guide/public-accessibility-logic.md]: Thanks for following up. Could you clarify what nomenclature you're referring to?

For additional context about why I've made this request:
 - The [documentation style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#headers) requests that level-3 headers use sentence case
 - `Lambda function` is used in AWS documentation (see [examples on this page](https://docs.aws.amazon.com/lambda/latest/dg/welcome.html))
 - `Lambda function` is what's used in other pages of the documentation site (see [examples on this page](https://docs.datadoghq.com/serverless/aws_lambda/installation/python/?tab=datadogcli))

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md (hunk 9 above)
Comment (buraizu) [reply content/en/security/cloud_security_management/guide/public-accessibility-logic.md:213]: Apologies for the back and forth here, but just requesting this minor update for the reasons outlined [here](https://github.com/DataDog/documentation/pull/22419#discussion_r1558047907).

Additionally, if you open the link, you'll find that `function` is not capitalized at the destination, either.

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for following up @Flowtter.

I've added some context to my previous requests, and requested some additional minor changes. Let me know if you have any questions.

Also, thanks for flagging the pattern of capitalization used on this page. I've opened a [separate PR](https://github.com/DataDog/documentation/pull/22579) to make the capitalization consistent across the doc, and compliant with our style guide.

File: content/en/security/cloud_security_management/guide/public-accessibility-logic.md (hunk 1 above)
Comment (buraizu) [reply content/en/security/cloud_security_management/guide/public-accessibility-logic.md]: Thanks for following up, I think I accidentally deleted my previous comment. Could you clarify what nomenclature you're referring to?

For additional context about why I'm making this request:
  - The [documentation style guide](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md#headers) requests sentence case for level-3 headers
  - AWS documentation mostly uses `Lambda function` (see [this page](https://docs.aws.amazon.com/lambda/latest/dg/getting-started.html) for examples)
  - Our documentation mostly uses `Lambda function` (see [this page](https://docs.datadoghq.com/serverless/aws_lambda/logs/?tab=serverlessframework) for examples)

Comment (Flowtter) [reply content/en/security/cloud_security_management/guide/public-accessibility-logic.md]: Thanks ! Following your PR, I made the changes that you suggested

**Review by buraizu — APPROVED** [review]

**Pull request merged**